tests/test_wheel.py

```diff
@@ -299,7 +302,11 @@ def test_limited_api(wheel_limited_api):
 @pytest.mark.skipif(MESON_VERSION < (1, 2, 99), reason='Meson version too old')
 @pytest.mark.xfail('__pypy__' in sys.builtin_module_names, reason='PyPy does not use special modules suffix for stable ABI')
 def test_limited_api_bad(package_limited_api, tmp_path):
-    with pytest.raises(mesonpy.BuildError, match='The package declares compatibility with Python limited API but '):
+    msg_start = 'The package declares compatibility with Python limited API but '
+    if NOGIL_BUILD:
+        msg_start = 'Free-threaded CPython does not support '
+
+    with pytest.raises(mesonpy.BuildError, match=msg_start):
```

Comment:
I think it would be clearer if the test would be simply skipped for nogil Python, otherwise it is harder to understand the purpose of the test.

Reply:
Sure, changed to skipping it.